service/src/worker.rs

```diff
@@ -85,8 +97,17 @@ where
 		Ok(())
 	}
 
+	/// Returns a list of peers. The address of the worker self is replaced by ws://localhost.
 	fn peers(&self) -> WorkerResult<Vec<EnclaveMetadata>> {
-		Ok(self.node_api.all_enclaves(None)?)
+		let mut peers = self.node_api.all_enclaves(None)?;
+		peers.iter_mut().for_each(|e| {
+			if e.url.trim_start_matches("ws://").trim_start_matches("wss://")
+				== self._config.worker_url()
+			{
+				e.url = format!("ws://localhost:{}", self._config.worker_rpc_port);
```

Comment:
hm - hardcoded ws? Is it not possible to replace 127.0.0.1 or whatever ip with localhost , instead of replacing it fully?

Reply:
I agree with @haerdib that this could and should be simplified. If a string contains our worker url, replace it with `localhost:port`. We don't have to care about the protocol (like `ws://` or `wss://`) at all (and shouldn't). Also I'd like to see some unit tests 😈